ocaml/typing/typecore.ml

```diff
@@ -10155,35 +10154,29 @@ let report_error ~loc env = function
       Location.errorf ~loc ~sub
         "@[This application is complete, but surplus arguments were provided afterwards.@ \
          When passing or calling a local value, extra arguments are passed in a separate application.@]"
-  | Param_mode_mismatch (s, mkind) ->
-      let print_error f (step, {Solver.left; Solver.right}) =
-        let actual, expected =
-          match (step : equate_step) with
-          | Left_le_right -> left, right
-          | Right_le_left -> right, left
-        in
-        Location.errorf ~loc
-          "@[This function takes a %a parameter,@ \
-           but was expected to take a %a parameter.@]"
-          f actual f expected
-      in begin
-      match mkind with
-      | Error (Comonadic Areality, e) ->
-          print_error Locality.Const.print (s, e)
-      | Error (Monadic Uniqueness, e) ->
-          print_error Uniqueness.Const.print (s, e)
-      | Error (Comonadic Linearity, e) ->
-          print_error Linearity.Const.print (s, e)
-      end
+  | Param_mode_mismatch (s, Error (ax, {left; right})) ->
+      let actual, expected =
+        match s with
+        | Left_le_right -> left, right
+        | Right_le_left -> right, left
+      in
+      Location.errorf ~loc
+        "@[This function takes a parameter at %a,@ \
+        but was expected to take a parameter at %a.@]"
```

Comment:
I think `at %a` isn't great, because of this error message (from one of the test files):
```
Error: This function when partially applied returns a value at once,
       but expected to be at many.
```
"returns a value at once" means "returns a value very soon", not "returns a value whose mode is `once`".

How about `which is %a` instead of `at %a`? (Other messages use modes with "to be", and I think it reads fine)

Reply:
Yeah `at once` has the advantage of close to the actual syntax `@ once`, but yeah it could be confusing for some mode constant names.